server/commands/subscriptionCreator.test.ts

```diff
@@ -0,0 +1,105 @@
+import { sequelize } from "@server/database/sequelize";
+import { Subscription, Event } from "@server/models";
+import { buildDocument, buildUser } from "@server/test/factories";
+import { flushdb } from "@server/test/support";
+import subscriptionCreator from "./subscriptionCreator";
+
+beforeEach(() => flushdb());
+
+describe("subscriptionCreator", () => {
+  const ip = "127.0.0.1";
+  const subscribedEvent = "documents.update";
+
+  it("should create subscription", async () => {
+    const user = await buildUser();
+
+    const document = await buildDocument({
+      userId: user.id,
+      teamId: user.teamId,
+    });
+
+    const subscription = await sequelize.transaction(async (transaction) =>
+      subscriptionCreator({
+        user: user,
+        documentId: document.id,
+        event: subscribedEvent,
+        ip,
+        transaction,
+      })
+    );
+
+    const event = await Event.findOne();
+
+    expect(subscription.documentId).toEqual(document.id);
+    expect(subscription.userId).toEqual(user.id);
+    expect(event?.name).toEqual("subscriptions.create");
+    expect(event?.modelId).toEqual(subscription.id);
+    expect(event?.actorId).toEqual(subscription.userId);
+    expect(event?.userId).toEqual(subscription.userId);
+    expect(event?.documentId).toEqual(subscription.documentId);
+  });
+
+  it("should not create another subscription if one already exists", async () => {
+    const user = await buildUser();
+
+    const document = await buildDocument({
+      userId: user.id,
+      teamId: user.teamId,
+    });
+
+    const subscription1 = await Subscription.create({
+      userId: user.id,
+      documentId: document.id,
+      event: subscribedEvent,
+    });
+
+    const subscription2 = await sequelize.transaction(async (transaction) =>
+      subscriptionCreator({
+        user: user,
+        documentId: document.id,
+        event: subscribedEvent,
+        ip,
+        transaction,
+      })
+    );
+
+    expect(subscription1.userId).toEqual(subscription2.userId);
+    expect(subscription1.documentId).toEqual(subscription2.documentId);
+    expect(subscription1.enabled).toEqual(subscription2.enabled);
+  });
+
+  it("should not enable subscription by overriding one that already exists in disabled state", async () => {
+    const user = await buildUser();
```

Comment:
Not sure if `subscriptions.create` should enable a disabled subscription. On one hand it makes a sort of sense, on the other we might encourage use of incorrect API semantics, which might cause a ton of headache in the future if we were to add other subscribeable events.

Reply:
@thenanyu not sure if you have opinions here?

The other way we could model this is and I'm kind of leaning towards now seeing how this is progressing is to just do standard create/delete and use a paranoid model so that before auto-subscribing we check if there is a previously deleted version of the subscription in the database. We can remove the `enabled` column then too…

Reply:
The other main reason to keep it is to make sure that when you edit a document again if you have previously disabled the subscription it shouldn't get automatically recreated as if you've never edited the document before.

Reply:
oh interesting! still think paranoid is the way to go. in this case deleted subs are acting as a tombstone, which is another valid usecase for paranoid